src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/TagHelpers/Producers/TagHelperProducer.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Razor.Language.TagHelpers.Producers;
+
+internal abstract class TagHelperProducer
+{
+    public abstract class FactoryBase : RazorEngineFeatureBase, IRazorEngineFeature, ITagHelperProducerFactory
```

Comment:
>FactoryBase [](http://example.com/codeflow?start=26&length=11)

Would it make sense for this to be genericized to the specific TagHelperProducer type it returns?

Reply:
I spent some time exploring that, but it didn't add value because everything ultimately calls through `ITagHelperProducerFactory`.